nix/Makefile

```
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Bao Project and Contributors. All rights reserved
+
+root_dir?=$(realpath ..)
```

I'd say the root dir by default should be:

```suggestion
root_dir?=$(realpath ../../)
```

Which, by our convention on how the ci repo should be included in the project's repos, should be the repo top-level directory and where all commands by default should run. And then, in the recipes:

```makefile
nix-shell:
	@cd $(root_dir) && nix-shell $(nix_dir)/shell.nix

nix-shell-run:
	@cd $(root_dir) && nix-shell $(nix_dir)/shell.nix --run "$(cmd)"
```